### PR TITLE
[WIP] Adds sqlalchemy pool values for nova.

### PR DESCRIPTION
**Pull request by dguitarbite**
Add sqlalchemy pool values to avoid hardcoding DB pool values for nova.

### CODE REVIEW

**Comment by toabctl** [comment]
I think you need a migration. 
What do you think about adding this to ```helpers.rb``` ? So we could set the pool values for all components (not only nova) globally. Or do we need this more fine grained on a per-component basis?

**Comment by dguitarbite** [comment]
@toabctl I saw the migration scripts for Cinder already present in there which should be reused IMO.

I am not sure if its a good idea to add this to helpers.rb to be reused, because these values should be independently customizable for each service. It may vary as per the architecture of deployment.

File: chef/data_bags/crowbar/bc-template-nova.json [hunk 1]
```diff
@@ -31,6 +31,10 @@
         "password": "",
         "user": "nova",
         "database": "nova"
+        "max_pool_size": 30,
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/bc-template-nova.json]: syntax error

**Comment by dirkmueller** [comment]
migration is missing

**Comment by dguitarbite** [comment]
Migrations added. Thanks Dirk.

**Comment by toabctl** [comment]
```033_add_configdrive.rb``` shouldn't be deleted

File: chef/data_bags/crowbar/migrate/nova/033_add_configdrive.rb [hunk 2]
```diff
@@ -1,13 +0,0 @@
...
-  end
-  return a, d
-end
-
-def downgrade(ta, td, a, d)
-  unless ta.has_key? "force_config_drive"
-    a.delete("force_config_drive")
-  end
-  return a, d
-end
```
Comment (aplanas) [inline chef/data_bags/crowbar/migrate/nova/033_add_configdrive.rb]: Why this migration is removed?

Comment (dguitarbite) [inline chef/data_bags/crowbar/migrate/nova/033_add_configdrive.rb]: Its a mistake, readding this file.

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 3]
```diff
@@ -2748,7 +2748,7 @@ connection=<%= @database_connection %>
 # (integer value)
 # Deprecated group/name - [DEFAULT]/sql_max_pool_size
 # Deprecated group/name - [DATABASE]/sql_max_pool_size
-#max_pool_size=<None>
+max_pool_size=<%= node[:nova][:max_pool_size] %>
```
Comment (dirkmueller) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: its [:nova][:db][:max_pool_size] on how you added it to the configuration..

File: chef/data_bags/crowbar/migrate/nova/034_add_sqlalchemy_pool_values.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+def upgrade(ta, td, a, d)
+  unless ta.has_key? "max_pool_size"
+    a["max_pool_size"] = ta["max_pool_size"]
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/migrate/nova/034_add_sqlalchemy_pool_values.rb]: the migration isn't checking the right keys. the keys are under db in your json update. Where did you want to put it really?

**Comment by dirkmueller** [comment]
-1 as it is, I suggest you test this before asking for reviews.. code review shouldn't find such glaring errors imho.

File: chef/data_bags/crowbar/migrate/nova/034_add_sqlalchemy_pool_values.rb (hunk 4 above)
Comment (dguitarbite) [inline chef/data_bags/crowbar/migrate/nova/034_add_sqlalchemy_pool_values.rb]: I wanted to up it under DB section but this migration checking code is unclear for me. Ill test it and send the correct code for review.

File: chef/data_bags/crowbar/migrate/nova/034_add_sqlalchemy_pool_values.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  end
+  unless ta["db"].has_key? "pool_timeout"
+    a["db"]["pool_timeout"] = ta["db"]["pool_timeout"]
+  end
+  return a, d
+end
+
+def downgrade(ta, td, a, d)
+  unless ta["db"].has_key? "max_pool_size"
+    a["max_pool_size"].delete("max_pool_size")
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/migrate/nova/034_add_sqlalchemy_pool_values.rb]: a['db']['...']

**Comment by dguitarbite** [comment]
Ill reopen it once I have tested it properly.

**Pull request closed without merging**